db.go

```diff
@@ -1751,7 +1744,6 @@ func (d *DB) Compact(start, end []byte, parallelize bool) error {
 	iStart := base.MakeInternalKey(start, InternalKeySeqNumMax, InternalKeyKindMax)
 	iEnd := base.MakeInternalKey(end, 0, 0)
 	m := (&fileMetadata{}).ExtendPointKeyBounds(d.cmp, iStart, iEnd)
```

Comment:
(not relevant to this PR so feel free to ignore) this seems like an abuse of a FileMetadata. Maybe we should use `KeyRange` instead?